src_base/xeve_args.c

```diff
@@ -0,0 +1,2313 @@
+    char videoformat[ARGS_OPT_VALUE_MAXLEN];
+    char range[ARGS_OPT_VALUE_MAXLEN];
+    char colorprim[ARGS_OPT_VALUE_MAXLEN];
+    char transfer[ARGS_OPT_VALUE_MAXLEN];
+    char master_display[ARGS_OPT_VALUE_MAXLEN];
+    char max_cll[ARGS_OPT_VALUE_MAXLEN];
+    char matrix_coefficients[ARGS_OPT_VALUE_MAXLEN];
+    int overscan_info_present_flag;
+    int overscan_appropriate_flag;
+    int chroma_loc_info_present_flag;
+    int chroma_sample_loc_type_top_field;
+    int chroma_sample_loc_type_bottom_field;
+    int neutral_chroma_indication_flag;
+    int field_seq_flag;
+    int timing_info_present_flag;
+    int num_units_in_tick;
+    int time_scale;
+    int fixed_pic_rate_flag;
+    int nal_hrd_parameters_present_flag;
+    int vcl_hrd_parameters_present_flag;
+    int low_delay_hrd_flag;
+    int pic_struct_present_flag;
+    int bitstream_restriction_flag;
+    int motion_vectors_over_pic_boundaries_flag;
+    int max_bytes_per_pic_denom;
+    int max_bits_per_mb_denom;
+    int log2_max_mv_length_horizontal;
+    int log2_max_mv_length_vertical;
+    int num_reorder_pics;
+    int max_dec_pic_buffering;
+};
+
+///////////////////////////////////////////////////////////////////////////////
+// Internal function declarations
+///////////////////////////////////////////////////////////////////////////////
+//
+static int args_search_long_key(ARGS_OPT * opts, const char * key);
+static int args_search_short_arg(ARGS_OPT * ops, const char key);
+static int args_read_value(ARGS_OPT * ops, const char * argv);
+static int args_parse_cfg(FILE* fp, ARGS_OPT* ops, int is_type_ppt);
+static int args_parse_cmd(int argc, const char * argv[], ARGS_OPT * ops, int * idx, char ** errstr);
+
+static int args_set_storage_for_option_by_long_key(ARGS_OPT * opts, char * key_long, void * var);
+static int args_set_storage_for_option_by_short_key(ARGS_OPT * opts, char * key, void * var);
+
+#define ARGS_SET_PARAM_VAR_KEY_LONG(opts, param, key_long) \
+    args_set_storage_for_option_by_long_key(opts, #key_long, (void*)&((param)->key_long))
+
+#define ARGS_SET_PARAM_VAR_KEY(opts, param, key) \
+    args_set_storage_for_option_by_short_key(opts, #key, (void*)&((param)->key))
+
+
+///////////////////////////////////////////////////////////////////////////////
+// Public API function implementations
+///////////////////////////////////////////////////////////////////////////////
+//
+ARGS_PARSER * xeve_args_create(void)
+{
+    ARGS_PARSER * args = NULL;
+    ARGS_OPT * opts = NULL;
+
+    args = malloc(sizeof(ARGS_PARSER));
+    if(args == NULL) goto ERR;
+    memset(args, 0, sizeof(ARGS_PARSER));
+
+    opts = malloc(sizeof(args_opt_table));
+    if (opts == NULL) goto ERR;
+    memcpy(opts, args_opt_table, sizeof(args_opt_table));
+    args->opts = opts;
+    args->num_option = ((int)(sizeof(args_opt_table)/sizeof(args_opt_table[0]))-1);
+    args->initialized = 0;
+
+    return args;
+
+ERR:
+    if(opts) {
+        free(opts);
```

Comment:
This line might be unreachable as opts value will be NULL in both the goto flow.